### PR TITLE
code-refactoring: streamline resource packages

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
> /kind enhancement
> /kind documentation
 /kind code-refactoring

**What does this PR do / why we need it**:
Currently each unique resource type defines its own  create/update/get/list/delete functions, which is repetitive and unnecessary, and opens the door up to inconsistent behavior

This PR adds a generalized resource management endpoint which will be used by all the resource packages under the hood
this allows reusing generic object management code and ensures consistent behavior across all object types while maintaining the individual resource endpoints 

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/workloads/deployment.go [hunk 1]
```diff
@@ -32,73 +31,70 @@ func newDeployment(objMeta metav1.ObjectMeta, spec appsv1.DeploymentSpec) *appsv
...
-
-	if err = client.Update(context.TODO(), deployment); err != nil {
-		return err
-	}
-	return nil
+	return resource.UpdateObject(deployment, client)
 }
 
+// DeleteDeployment deletes the Deployment with the given name and namespace using the provided client.
+// It ignores the "not found" error if the Deployment does not exist.
```
Comment (svghadi) [inline pkg/workloads/deployment.go:66]: I find it a bit confusing. Can we rephrase this? Maybe something like 
```suggestion
// It returns the "not found" error if the Deployment does not exist.
```

**Comment by svghadi** [comment]
Had a initial look, left a comment. Please let me know when it is ready for final review, need it for #1063.

File: pkg/workloads/deployment.go (hunk 1 above)
Comment (jaideepr97) [reply pkg/workloads/deployment.go]: Thanks, I removed all the comments that said "not found" errors are ignored

**Review by svghadi — APPROVED** [review]
LGTM. Thanks

**Pull request merged**